src/Component/RuleTable/RuleTable.tsx

```diff
@@ -382,6 +385,53 @@ export class RuleTable extends React.Component<RuleTableProps, RuleTableState> {
     );
   }
 
+  ruleOrderDownRenderer = (record: RuleRecord) => {
+    const disableOrderDown = record.key === this.props.rules.length - 1;
+    return this.ruleOrderRenderer(record.key, true, disableOrderDown);
+  }
+
+  ruleOrderUpRenderer = (record: RuleRecord) => {
+    const disableOrderUp = record.key === 0;
+    return this.ruleOrderRenderer(record.key, false, disableOrderUp);
+  }
+
+  ruleOrderRenderer = (ruleIndex: number, moveDown: boolean, disabled: boolean) => {
```

Comment:
This should be a seperated component.

Reply:
Not a lot sub components are separated in the RuleTable component  I just followed the line in this component. :wink:
But I can start by separating the re-order buttons.